### ISSUE TITLE
Anomalies in CN parsing of `for` loops

### ISSUE THREAD

**Issue by septract** [rems-project/cerberus#232]
CN will not parse a `for` loop with both (1) an invariant, and (2) a variable declaration. 

That is, parsing fails on this example: 
```
int for_errorcase() 
{
  int acc = 0; 
  for(int i = 0; i <= 10; i++)    // <-- This example contains both a variable declaration 
  /*@ inv 0i32 <= i; i <= 10i32;  // <--  ...and an invariant 
          acc <= 10i32 @*/ 
  {
    acc = i; 
  }; 
  return acc;
}
```

However, parsing succeeds on both of the following examples: 
```
int for_declaration() 
{
  int acc = 0; 
  for(int i = 0; i <= 10; i++)
  {
    acc = i; 
  }; 
  return acc;
}

int for_invariant() 
{
  int acc = 0; 
  int i; 
  for(i = 0; i < 10; i++)
  /*@ inv 0i32 <= i; i <= 10i32;
          acc <= 10i32 @*/
  {
    acc = i; 
  }; 
  return acc;
}
```

Suggested resolution from @yav: desugar `for (int i = ..) INV BODY` into `{ int i; for (i = ..) INV BODY }`

**Comment by dc-mak** [comment]
oops - should be fixed by c10d117

**Comment by septract** [comment]
Confirmed - looks like this fixes the bug. Thanks!
